CorpusCallosum/cc_visualization.py

```diff
@@ -0,0 +1,152 @@
+import argparse
+from pathlib import Path
+
+import numpy as np
+
+from CorpusCallosum.data.fsaverage_cc_template import load_fsaverage_cc_template
+from CorpusCallosum.shape.cc_mesh import CC_Mesh
+
+
+def options_parse() -> argparse.Namespace:
+    """Parse command line arguments for the visualization pipeline."""
+    parser = argparse.ArgumentParser(description="Visualize corpus callosum from template files.")
+    parser.add_argument("--contours", type=str, required=False, help="Path to contours.txt file", default=None)
+    parser.add_argument("--thickness", type=str, required=True, help="Path to thickness_values.txt file")
+    parser.add_argument(
+        "--measurement_points",
+        type=str,
+        required=True,
+        help="Path to measurement points file containing the original vertex indices where thickness was measured",
+    )
+    parser.add_argument("--output_dir", type=str, required=True, help="Directory for output files")
+    parser.add_argument("--resolution", type=float, default=1.0, help="Resolution in mm for the mesh")
+    parser.add_argument(
+        "--smooth_iterations", type=int, default=1, help="Number of smoothing iterations to apply to the mesh"
+    )
+    parser.add_argument(
+        "--colormap",
+        type=str,
+        default="red_to_yellow",
+        choices=["red_to_blue", "blue_to_red", "red_to_yellow", "yellow_to_red"],
+        help="Colormap to use for thickness visualization",
+    )
+    parser.add_argument(
+        "--color_range",
+        type=float,
+        nargs=2,
+        default=None,
+        metavar=("MIN", "MAX"),
+        help="Optional fixed range for the colorbar (min max)",
+    )
+    parser.add_argument("--legend", type=str, default="Thickness (mm)", help="Legend for the colorbar")
+    parser.add_argument("--twoD", action="store_true", help="Generate 2D visualization instead of 3D mesh")
+
+    args = parser.parse_args()
+
+    # Create output directory if it doesn't exist
+    Path(args.output_dir).mkdir(parents=True, exist_ok=True)
+
+    return args
+
+
+def main(
+    contours_path: str | Path | None,
+    thickness_path: str | Path,
+    measurement_points_path: str | Path,
+    output_dir: str | Path,
+    resolution: float = 1.0,
+    smooth_iterations: int = 1,
+    colormap: str = "red_to_yellow",
+    color_range: tuple[float, float] | None = None,
+    legend: str | None = None,
+    twoD: bool = False,
+) -> None:
```

Comment:
Return of `main` should be 0 or str
```suggestion
) -> Literal[0] | str:
```